app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -39,6 +39,24 @@ def create
     redirect_to edit_chat_channel_membership_path(membership)
   end
 
+  def join_channel
+    membership_params = params[:chat_channel_membership]
+    chat_channel = ChatChannel.find(membership_params[:chat_channel_id])
+    user = User.find_by(id: membership_params[:user_id])
+    existing_membership = ChatChannelMembership.find_by(user_id: user.id, chat_channel_id: chat_channel.id)
+    if existing_membership.present? && %w[active joining_request].exclude?(existing_membership.status)
+      success = existing_membership.update(status: "joining_request", role: "member") ? true : false
+    else
+      membership = ChatChannelMembership.new(user_id: user.id, chat_channel_id: chat_channel.id, role: "member", status: "joining_request")
+      success = membership.save ? true : false
```

Comment:
Same as above, save will return true or false already.

Reply:
Thanks for pointing this out. Modified